components/fxa-client/src/http_client.rs

```diff
@@ -24,6 +34,12 @@ pub trait FxAClient {
         refresh_token: &str,
         scopes: &[&str],
     ) -> Result<OAuthTokenResponse>;
+    fn refresh_token_with_session_token(
+        &self,
+        config: &Config,
+        session_token: &[u8],
+        scopes: &[&str],
+    ) -> Result<OAuthTokenResponse>;
```

Comment:
Should this be name `oauth_token_with_session_token` for consistency with the other methods above? I think it's doing the same sort of grant you'd do with e.g. a `code` and obtaining the same response, right?

Reply:
name collision with another `oauth_token_with_session_token` we have, have to pick something else for now